### ISSUE TITLE
Problems hydrating previously unhydrated PortfolioItem objects

### ISSUE THREAD

**Issue by patknight** [RallyTools/RallyRestToolkitForPython#48]
BIG REWORDING AS I THINK MY ORIGINAL REPORT WAS CONFUSINGLY WORDED.

These problems arose when WSAPI 2.0 started requiring the full hierarchical entity type name of portfolio items in the REST API instead of the sub type name (i.e. requiring PortfolioItem/Feature instead of Feature.) This is handled fine in the initial request to get an object, but doesn't work when pyral needs to hydrate a previously unhydrated object. This is because the code to hydrate the object that builds a get request  doesn't handle the "/" in the middle of portfolio entity type names.

There are two issues:
1) In entity.py `Persistable.__getattr__` incorrectly extracts portolio item entity names when splitting the self._ref. To construct the entity name to send to the Rally API server it needs to cope with a '/' within entity types ('PortfolioItem/Feature' vs 'HierarchicalRequirement')

This is addressed by first splitting on `WS_API_VERSION + '/'` to get a string containing just the entity name and oid before further splitting that to extract those fields.

2) Although the server wants the entity type to be e.g. 'PortfolioItem/Feature' in the request, it uses 'Feature' as the key in the response content. However, the Pyral class for the unhydrated feature object (in this case) is 'PortfolioItem' so we get a KeyError when we try to extract the response content.

This is addressed by catching the KeyError on 'PortfolioItem', then looping through PORTFOLIO_ITEM_SUB_TYPES, re-raising the original exception if we don't find the contents.

Here's a file that shows the problem (if you have access to my workspace and projects - otherwise just put in your own)
```python
import pyral
import metadata


class RallyError(Exception):
    pass


_rally_instance = None


def get_rally_instance():
    global _rally_instance

    if _rally_instance is None:
        _rally_instance = pyral.Rally(
            server=metadata.rally_auth[u"server"],
            user=metadata.rally_auth[u"user"],
            password=metadata.rally_auth[u"password"],
            workspace=metadata.rally_auth[u"workspace"],
            project=metadata.rally_auth[u"project"])

    return _rally_instance

rally = get_rally_instance()

unhydrated = list(rally.get("PortfolioItem/Feature", fetch=False,
                            query='FormattedID = "F226"'))[0]
hydrated = list(rally.get("PortfolioItem/Feature", fetch=True,
                         query='FormattedID = "F226"'))[0]
print hydrated.FormattedID # this works
print unhydrated.FormattedID # this throws an unknown OID 422 exception
```

**Comment by patknight** [comment]
This appears to be a duplicate of issue #41.

**Comment by patknight** [comment]
I've added pull request #53 with my proposed fix. Pull request #41 from ksato9700 has an alternative approach which I found after I'd coded mine. Please choose whichever you think is more in keeping with the existing code base.

**Comment by klehman-rally** [comment]
Fixed in 1.1.1.  The code to address this issue was a result of an internally reported defect was held for some period of time to bundle several changes as part of 1.1.1.  I'm going to close the associated pull request #53 .  Feel free to re-submit that PR if 1.1.1 does not suitably address the problem.

**Comment by patknight** [comment]
I've just confirmed that 1.1.1 does fix the problem. Thank-you.
